SoftLayer/CLI/virt/create.py

```diff
@@ -133,11 +133,9 @@ def _parse_create_args(client, args):
     if args.get('vlan_private'):
         data['private_vlan'] = args['vlan_private']
 
-    if args.get('subnet_public'):
-        data['public_subnet'] = args['subnet_public']
+    data['public_subnet'] = args.get('subnet_public', None)
```

Comment:
I tried to implement the switcher option for this method, but we get the ¨args¨ parameter, it's not a specific data. I was implementing like this:

    switcher = {
        args.get('san'): attribute_san(args, data),
        args.get('os'): attribute_os(args, data),
        args.get('image'): attribute_image(args, client, data),
        args.get('datacenter'): attribute_datacenter(args, data),
        args.get('network'): attribute_network(args, data),
        args.get('postinstall'): attribute_postinstall(args, data),
        args.get('key'): attribute_key(args, client, data),
        args.get('vlan_public'): attribute_vlan_public(args, data),
        args.get('vlan_private'): attribute_vlan_private(args, data),
        args.get('subnet_public'): attribute_subnet_public(args, data),
        args.get('subnet_private'): attribute_subnet_private(args, data),
        args.get('public_security_group'): attribute_public_security_group(args, data),
        args.get('private_security_group'): attribute_private_security_group(args, data),
        args.get('tag'): attribute_tag(args, data),
        args.get('host_id'): attribute_host_id(args, data),
    }

    return data


def attribute_host_id(args, data):
    if args.get('host_id'):
        data['host_id'] = args['host_id']


def attribute_tag(args, data):
    if args.get('tag'):
        data['tags'] = ','.join(args['tag'])


def attribute_private_security_group(args, data):
    if args.get('private_security_group'):
        priv_groups = args.get('private_security_group')
        data['private_security_groups'] = [group for group in priv_groups]
..........................

In this case the attribute switches will not be use, because in each method each data from args will be add to the data attribute.
That's why I used the first option you suggested.